src/app/components/core/north/north-task-modal/north-task-modal.component.ts

```diff
@@ -234,17 +234,16 @@ export class NorthTaskModalComponent implements OnInit, OnChanges {
   }
 
   public saveScheduleFields(form: NgForm) {
-    if (!form.dirty && !form.touched) {
-      this.toggleModal(false);
-      return false;
-    }
-
     if (this.isFilterDeleted) {
       this.deleteFilter();
     }
     if (this.isFilterOrderChanged) {
       this.updateFilterPipeline(this.filterPipeline);
     }
+    if (!form.dirty && !form.touched) {
```

Comment:
we need to put a comment regrading https://angular.io/guide/form-validation#why-check-dirty-and-touched